### DISCUSSION TITLE
GloBrite Color Changing lights

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#550] [Nixie]
Hi folks.  I have 3 GloBrite color changing lights.  They are each connected to their own circuit.  Is it possible to adjust the on-off frequency for switching color modes?  I believe I need to increase the time between on-off, ie reduce frequency, as my lights turn off during color change, but come back to the same color it was prior to change.

**Comment by rstrouse** [comment]
I assume you have these hooked to either a REM relay or GPIO.  If you do there is an On and Off delay to add time to the sequence when you define the relay.  The same is true for GPIO.

**Reply by orrious** [reply]
Yeah, it's going through REM.  Do you have the standard timing that the REM delays add to?

**Reply by rstrouse** [reply]
These seem to work fine for IntelliBrite at 100ms.  ColorLogic seems to require 200ms.   You will see the sequence flash in the log when it gets sent.  Then you can extend that timing in REM.   This timing is what was determined by others switching IntelliBrite but GloBrite may require more time.  If you find a good number that reliably switches GloBrite then let me know so we can change `controller/nixie/circuits/Circuits.ts` line 163 & 164.

**Reply by orrious** [reply]
Tried timing my lights last night and the timing is off on the first pulse.  Is there any way to actually see the timing on the pulses other than getting out my oscilloscope? 

The first pulse doesn't appear to be in time with the rest of the pulses.  It doesn't matter what I set the delay to, light flickers off/on/ff, stays off and then goes back to the original setting.  If I manually press the button on and off, mode changes fine.  I've even tried increasing the timing to 500ms.

**Reply by rstrouse** [reply]
What circuit type do you have set for the light circuit?

**Reply by orrious** [reply]
I've tried them as GloBrite and Intellibrite, both have the same result.

**Reply by rstrouse** [reply]
The first pulse turns the light off.  It has to do this because Pentair lights start their sequence from off.  That is the difference for the first pulse.  If you select a theme the sequence will appear in the njsPC log.

**Reply by rstrouse** [reply]
Btw are you switching the low voltage side of the transformer?  There have been several folks that have this working.  Something has to be odd about this.

**Reply by orrious** [reply]
Both sides of the transformer are switched with SSR.  The mains side (HV) comes on at dusk and off at 2am.  All my other landscape lights also use this transformer for 12vac.   The Globrites each have their own SSR as well.  This setup has been working for years for me with my own board.  Also, I've never differentiated the first pulse, neither on my board nor manually testing just by turning the light on and off manually.  I believe the issue is the timing of that first pulse.  Is it possible to make all the pulse on/off times exactly the same?

**Reply by rstrouse** [reply]
The first command to the light is not necessarily a pulse.  It is setting up the light controller to start accepting commands.  It needs to be a bit longer (~1s if the light is off already and ~2s if the light is currently on).  Is it possible that your dual relay has the main relay off when it is pulsing the other relay?
